tests/components/selectBox/index.test.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import SelectBox from '../../../app/js/components/selectBox';
+
+let props;
+let mountedComponent;
+
+const getComponent = () => {
+  if (!mountedComponent) {
+    mountedComponent = shallow(<SelectBox {...props} />);
+  }
+  return mountedComponent;
+};
+
+describe('Component: SelectBox', () => {
+  beforeEach(() => {
+    props = {
+      name: 'sets',
+      items: [{ value: 'Choose a set to add', id: 0 }, { value: 'Mineral', id: 1 }],
+      selectedItem: {},
+      onChange: jest.fn(),
+    };
+    mountedComponent = undefined;
+  });
+
+  it('renders properly', () => {
+    const component = getComponent();
+    expect(component).toMatchSnapshot();
+  });
+
+  it('should call dropDown()', () => {
+    const renderedComponent = getComponent().instance();
+    const dropDown = jest.spyOn(renderedComponent, 'dropDown');
+    renderedComponent.dropDown();
+    expect(renderedComponent.dropDown).toBeCalled();
```

Comment:
I think one of the objectives of this sprint is to do away with writing tests this way. 
calling a components instance method and using jest's `spyOn` method to assert it was called. this doesn't convey proper context about the behavior of the component.

A basic rule of thumb when writing tests for React components is to write your tests with respect to the component's props and state.

please take out time to read and understand this blogpost ..

https://medium.freecodecamp.org/the-right-way-to-test-react-components-548a4736ab22

😄

Reply:
This has been refactored thanks for the tip